projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/experiment/Experiment.java

```diff
@@ -0,0 +1,340 @@
+    addVariableToDesign(variableName, levels).ifFailure(
+        e -> {
+          if (e instanceof ExperimentalVariableExistsException) {
+            // at this point we know there is a variable with the name `specimen`, so we only need to add the levels
+            for (ExperimentalValue level : levels) {
+              experimentalDesign.addLevelToVariable(variableName, level).ifFailure(
+                  e2 -> {
+                    //FIXME what exception to throw here?
+                    //   we know that the experimental variable must be defined at that point so we do not expect this exception
+                    throw new RuntimeException(
+                        "could not add level " + level + " to variable "
+                            + variableName, e2);
+                  });
+            }
+          } else if (e instanceof IllegalArgumentException) {
+            //FIXME what exception to throw here?
+            throw new RuntimeException(e);
+          }
+        }
+    );
+  }
+
+  /**
+   * Creates a new condition and adds it to the experimental design. A successful operation is
+   * indicated in the result, which can be verified via {@link Result#isSuccess()}.
+   * <p>
+   * <b>Note</b>: {@link Result#isFailure()} indicates a failed operation.
+   * {@link Result#exception()} can be used to determine the cause of the failure.
+   * <ul>
+   *   <li>If a condition with the provided label or the same variable levels already exists, the creation will fail with an {@link ConditionExistsException} and no condition is added to the design.
+   *   <li>If the {@link VariableLevel}s belong to variables not specified in this experiment, the creation will fail with an {@link IllegalArgumentException}
+   * </ul>
+   *
+   * @param conditionLabel a declarative and unique name for the condition in scope of this
+   *                       experiment.
+   * @param levels         at least one value for the variable
+   * @return a {@link Result} object containing the {@link ConditionLabel} or containing a
+   * declarative exceptions.
+   */
+  public Result<ConditionLabel, Exception> addConditionToDesign(String conditionLabel,
+      VariableLevel[] levels) {
+    Arrays.stream(levels).forEach(Objects::requireNonNull);
+
+    for (VariableLevel level : levels) {
+      if (!experimentalDesign.isVariableDefined(level.variableName().value())) {
+        return Result.failure(new IllegalArgumentException(
+            "There is no variable " + level.variableName().value() + " in this experiment."));
+      }
+    }
+    boolean areAllLevelsFromDefinedVariables = Arrays.stream(levels)
+        .allMatch(it -> experimentalDesign.isVariableDefined(it.variableName().value()));
+    if (!areAllLevelsFromDefinedVariables) {
+      return Result.failure(
+          new IllegalArgumentException(
+              "Not all levels are from variables defined in this experiment"));
+    }
+
+    try {
```

Comment:
That is very hard to understand, please try to extract parts into own methods. Also, what does defined variables mean? You check for new levels or unknown levels. If you find unknown levels or new levels, then you cant create the condition.